### PR TITLE
LPD-37535 Ids of all inactive groups are added to search query in manual collection searches

### PR DESCRIPTION
**Pull request by AlmirFe**
https://liferay.atlassian.net/browse/LPD-37535

### CODE REVIEW

**Comment by AlmirFe** [comment]
ci:test:sf

**Comment by AlmirFe** [comment]
ci:help

**Comment by AlmirFe** [comment]
ci:test:relevant

**Comment by AlmirFe** [comment]
ci:test:relevant

File: modules/apps/asset/asset-list-service/src/main/java/com/liferay/asset/list/internal/asset/entry/provider/AssetListAssetEntryProviderImpl.java [hunk 1]
```diff
@@ -317,37 +320,39 @@ private long[] _filterAssetCategoryIds(long[] assetCategoryIds) {
...
-				if ((assetEntry == null) || !assetEntry.isVisible()) {
-					return false;
-				}
+			AssetEntry assetEntry = entry.getValue();
 
-				AssetRendererFactory<?> assetRendererFactory =
-					AssetRendererFactoryRegistryUtil.
-						getAssetRendererFactoryByClassName(
-							assetEntry.getClassName());
+			if ((assetEntry == null) || !assetEntry.isVisible()) {
```
Comment (joshuacords) [inline modules/apps/asset/asset-list-service/src/main/java/com/liferay/asset/list/internal/asset/entry/provider/AssetListAssetEntryProviderImpl.java:333]: Move this null/visible check to ln 468 and don't even add "bad" entries to the map.

File: modules/apps/asset/asset-list-service/src/main/java/com/liferay/asset/list/internal/asset/entry/provider/AssetListAssetEntryProviderImpl.java [hunk 2]
```diff
@@ -411,34 +416,32 @@ private BooleanClause[] _getAssetCategoryIdsBooleanClauses(
 		};
 	}
 
-	private List<AssetListEntryAssetEntryRel> _getAssetListEntryAssetEntryRels(
-		AssetListEntry assetListEntry, long[] segmentsEntryIds) {
+	private Map<AssetListEntryAssetEntryRel, AssetEntry>
```
Comment (joshuacords) [inline modules/apps/asset/asset-list-service/src/main/java/com/liferay/asset/list/internal/asset/entry/provider/AssetListAssetEntryProviderImpl.java:419]: The Map seems like overkill, don't we really only need the priority and groupIds?

File: modules/apps/asset/asset-list-service/src/main/java/com/liferay/asset/list/internal/asset/entry/provider/AssetListAssetEntryProviderImpl.java (hunk 1 above)
Comment (joshuacords) [reply modules/apps/asset/asset-list-service/src/main/java/com/liferay/asset/list/internal/asset/entry/provider/AssetListAssetEntryProviderImpl.java:333]: Same with the null AssetRendererFactory check.

**Review by joshuacords — CHANGES_REQUESTED** [review]
I would like you to refactor to see about removing the Map, or at least simplifying the types. It isn't really even being used as a map (as a way to enter a key and recall a value) but just storage.

I'm up for pair programming this if you could use more direction.

**Comment by AlmirFe** [comment]
ci:test:sf

**Comment by joshuacords** [comment]
LGTM! :rocket:

**Comment by joshuacords** [comment]
ci:test:relevant

**Comment by AlmirFe** [comment]
ci:forward

**Pull request closed without merging**